### PR TITLE
Handle replacing unknown node inside group or subflow

### PR DESCRIPTION
**Pull request by knolleary**
This fixes two issues related to replacing unknown nodes in the editor.

One reported https://discourse.nodered.org/t/group-does-not-show-styling-until-browser-refresh-made/69078 - the other discovered whilst fixing the first one.

Background: when importing a flow, any unrecognised types are replaced by a special `unknown` node. If that unrecognised type gets registered later on (either some very async work was done before the type registered, or the user had to install it via the Palette Managed), the editor would swap out the unknown nodes for the 'real' node.

This PR fixes two scenarios that were not being handled properly:

1. if the node was in a group, the group was left with the `unknown` node still referenced in its internal list of nodes

   This fix ensures the `unknown` node is swapped out for the proper node

2. if the node was in a subflow, it would get 'moved' to the current flow

    The `importNodes` code has been updated to handle reimporting a node back into a subflow

### CODE REVIEW

**Comment by knolleary** [comment]
Merging is blocked until #3922 is merged and this branch rebased to pickup that fix.

**Review by Steve-Mcl — APPROVED** [review]
Looks sane.

**Pull request merged**
